CUDADataFormats/BeamSpot/src/BeamSpotCUDA.cc

```diff
@@ -0,0 +1,9 @@
+#include "CUDADataFormats/BeamSpot/interface/BeamSpotCUDA.h"
+
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+
+BeamSpotCUDA::BeamSpotCUDA(Data const* data_h, cudaStream_t stream) {
+  data_d_ = cudautils::make_device_unique<Data>(stream);
+  cudaCheck(cudaMemcpyAsync(data_d_.get(), data_h, sizeof(Data), cudaMemcpyHostToDevice, stream));
```

Comment:
@makortel @VinInn could you remind me why we have the `cudaMemcpyAsync` in the dataformat itself, rather than in the relevant EDProducer ?

Reply:
The division of work between the EDProducer and the data format class is somewhat arbitrary. This way the details of the (device) memory allocations are fully controlled by the data format class, and the object can be constructed+inserted into the event within `Event::emplace()`.